public/app/plugins/datasource/tempo/datasource.ts

```diff
@@ -202,24 +203,80 @@ export class TempoDatasource extends DataSourceWithBackend<TempoQuery, TempoJson
         return of({ error: { message: error instanceof Error ? error.message : 'Unknown error occurred' }, data: [] });
       }
     }
+
     if (targets.traceql?.length) {
       try {
-        const appliedQuery = this.applyVariables(targets.traceql[0], options.scopedVars);
-        const queryValue = appliedQuery?.query || '';
-        const hexOnlyRegex = /^[0-9A-Fa-f]*$/;
-        // Check whether this is a trace ID or traceQL query by checking if it only contains hex characters
-        if (queryValue.trim().match(hexOnlyRegex)) {
-          // There's only hex characters so let's assume that this is a trace ID
-          reportInteraction('grafana_traces_traceID_queried', {
-            datasourceType: 'tempo',
-            app: options.app ?? '',
-            grafana_version: config.buildInfo.version,
-            hasQuery: queryValue !== '' ? true : false,
-          });
+        const groupBy = targets.traceql.find((t) => this.hasGroupBy(t));
+        if (groupBy) {
+          subQueries.push(
+            this.handleMetricsSummary(groupBy, this.applyVariables(groupBy, options.scopedVars)?.query || '', options)
+          );
+        }
 
-          subQueries.push(this.handleTraceIdQuery(options, targets.traceql));
-        } else {
-          reportInteraction('grafana_traces_traceql_queried', {
+        const traceqlTargets = targets.traceql.filter((t) => !this.hasGroupBy(t));
```

Comment:
I assume this diff is mainly because of reordering the IFs here right? Or did something change apart from the `if (groupBy) {` part?

Reply:
Yep this is basically reordering/update because of the group by part.